subsys/disk/disk_access_spi_sdhc.c

```diff
@@ -16,20 +18,28 @@ LOG_MODULE_REGISTER(sdhc_spi, CONFIG_DISK_LOG_LEVEL);
 #include "disk_access_sdhc.h"
 
 /* Clock speed used during initialisation */
-#define SDHC_SPI_INITIAL_SPEED 400000
+#define SDHC_SPI_INITIAL_SPEED KHZ(400)
 /* Clock speed used after initialisation */
-#define SDHC_SPI_SPEED 4000000
+#define SDHC_SPI_DEFAULT_SPEED MHZ(25)
 
 #if !DT_NODE_HAS_STATUS(DT_INST(0, zephyr_mmc_spi_slot), okay)
 #warning NO SDHC slot specified on board
 #else
+enum sdhc_spi_cfg {
+	SDHC_SPI_INIT_CFG,
+	SDHC_SPI_DEFAULT_SPEED_CFG,
+	/* --- */
+	SDHC_CFG_COUNT,
+};
+
 struct sdhc_spi_data {
 	struct device *spi;
-	struct spi_config cfg;
+	struct spi_config spi_cfgs[SDHC_CFG_COUNT];
```

Comment:
These structures are filled only with values that are known at the build time, so they could be extracted to the constant `config_info` structure for the device (which is currently not used at all; [see](https://github.com/zephyrproject-rtos/zephyr/blob/8f7db05c6f414b219b33a1fdf029b8b84b7d3b1c/subsys/disk/disk_access_spi_sdhc.c#L939) the `NULL` in the `DEVICE_AND_API_INIT` call) and use static initialization. This would save some RAM and flash.

Reply:
Agreed; there should be a file-local type and object like:
```
struct spi_sdhc_config {
   struct spi_config init_cfg;
   struct spi_config oper_cfg;
} sdhc_spi_cfg_0;
```
with a `static const` instance that's passed via `DEVICE_AND_API_INIT`.

Then in this structure there should just be:
```
    const struct spi_config *spi_cfg;
```
that  is assigned to either `&config->init_cfg` or `&config->oper_cfg` at the points where `disk->status` is changed to `UNINIT` or `OK`.

(just as @nzmichaelh recommended, now that I reach that comment.)